### DISCUSSION TITLE
Cluster Setup - Resource deadlock avoided

### DISCUSSION THREAD

**Discussion by ghost** [ory/hydra#2929] [Q&A] [answered]
We are running an Ory Hydra cluster (3 nodes) with a MariaDB cluster (version: 10.3.29) with an HAProxy in between.

The cluster is monitored with Elastic APM. Here we see more and more the following error messages: 

- `Error 1180: Got error 35 "Resource deadlock avoided" during COMMIT"`
- `Error 1213: Deadlock: wsrep aborted transaction`

As far as I have seen, the error messages come from the MariaDB. Is there any configuration we can do here?
After some research, we are pretty sure, that we could mitigate this by only writing to a single node. 

However, this is something that would need to be implemented in hydra?

**Comment by vinckr** [comment, marked as answer]
Hello @markus-ki 

Is it this issue: https://www.ory.sh/hydra/docs/limitations/#mysql--56--mariadb, more detail: https://github.com/ory/hydra/issues/377 ?

**Reply by vinckr** [reply]
Hey @markus-ki , just checking in if your problem still persists. 
I will mark this as answered for now, but feel free to come back to the discussion any time.
Thanks a ton :bee:

**Comment by edqartwizaru** [comment]
Hi can you help me how to setup ory/hydra in Kubernetes
 cluster
